org.eclipse.jdt.ls.product/scripts/jdtls.py

```diff
@@ -103,7 +107,7 @@ def main(args):
 			"--add-opens", "java.base/java.util=ALL-UNNAMED",
 			"--add-opens", "java.base/java.lang=ALL-UNNAMED"] \
 			+ known_args.jvm_arg \
-			+ ["-jar", jar_path,
+			+ ["-jar", str(jar_path),
```

Comment:
I would probably do the same thing in `return plugins_dir / launcher`. Just make it `str(return plugins_dir / launcher)` and that should ensure everything is a string.

Reply:
Same as above, wasn't sure the intended result was to return a string always. I've moved it to surround the return of `find_equinox_launcher`. I've also done the same for `get_shared_config_path` with the same rationale.